### PR TITLE
nrf_modem: release v1.4.0

### PR DESCRIPTION
**Pull request by lemrey**
Modem library release v1.4.0.

_commits to be fixed-up_

### CODE REVIEW

**Comment by balaji-nordic** [comment]
Dont we need an some explanations for the new `nrf_modem_os*` functions in `ug_nrf_modem_porting_os.rst`?

**Comment by lemrey** [comment]
Thanks @balaji-nordic, I have updated the porting guide with a brief description for each function.

**Comment by lemrey** [comment]
Squashed all commits

**Review by rlubos — CHANGES_REQUESTED** [review]
Why did the lib size increase that much?
```
BIN +8.37 MB (5700%) nrf_modem/lib/cortex-m33/hard-float/libmodem.a
````
I don't think we should merge it like this.

File: nrf_modem/doc/CHANGELOG.rst [hunk 1]
```diff
@@ -9,29 +9,56 @@ Changelog
...
+* The GNSS socket has been removed.
+* nrf_errno errno values have been aligned with those of newlibc.
+* The :ref:`Modem API <nrf_modem_api>` (:file:`nrf_modem.h`) has been updated to return negative errno values on error.
+* The :ref:`Full Modem DFU API <nrf_modem_full_dfu_api>` (:file:`nrf_modem_full_dfu.h`) has been updated to return negative errno values on error.
+* The :ref:`GNSS API <nrf_modem_gnss_api>` (:file:`nrf_modem_gnss.h`) has been updated to return negative errno values on error.
+* The :c:func:`nrf_modem_gnss_init` and :c:func:`nrf_modem_gnss_deinit` functions have been removed.
+* Added the GNSS velocity estimate validity bit ``NRF_MODEM_GNSS_PVT_FLAG_VELOCITY_VALID``.
+* Added the GNSS delete bitmask ``NRF_MODEM_GNSS_DELETE_GPS_TOW_PRECISION`` for time-of-week precision estimate.
+* Added support for several new fields in the GNSS PVT notification.
+* Added support for GNSS A-GPS data need query.
```
Comment (umapraseeda) [inline nrf_modem/doc/CHANGELOG.rst:25]:
```suggestion
* Added support for checking when A-GPS data is needed.
```

File: nrf_modem/doc/pdn.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The Modem library supports selecting which Packet Data Network (PDN) to use on a network socket and for DNS queries.
+The configuration of Packet Data Protocol (PDP) contexts, and the activation of PDN connections are not handled by the Modem library.
+To configure PDP contexts and activate PDN connections, the application must use the nRF9160 modem Packet Domain AT commands.
+
+When performing network operations on any PDN, the application has the responsibility to ensure that the PDN connection is available.
+For more information about how to configure PDP contexts, activate PDN connections, and determine their state, see the `nRF9160 modem Packet Domain AT commands`_ documentation.
+
+
+Configuring a network socket to a use a PDN
```
Comment (eivindj-nordic) [inline nrf_modem/doc/pdn.rst:18]: "...to use a..."

File: nrf_modem/include/nrf_modem_gnss.h [hunk 3]
```diff
@@ -479,6 +512,38 @@ struct nrf_modem_gnss_agps_data_frame {
 	uint32_t data_flags;
 };
 
+/** @brief The number of GPS satellites. */
+#define NRF_MODEM_GNSS_NUM_GPS_SATELLITES 32
+
+/** @brief A-GPS data need and expiry times.
```
Comment (eivindj-nordic) [inline nrf_modem/include/nrf_modem_gnss.h:518]: This sentence is unclear.

**Review by lemrey — COMMENTED** [review]
@rlubos thanks for checking, the change is intentional. The increased size is because the library now contains debugging symbols.
There is a line in the release notes about that.

**Comment by tokangas** [comment]
> Would A-GPS data expiry work, @tokangas?

@lemrey Yes, that's fine with me.

**Review by umapraseeda — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
Lifting my -1, but IMO we should really reconsider our strategy regarding library releases in the future. Putting dozens of megabytes on each library release will increase the repository size greatly.

**Pull request merged**